MCprep_addon/spawner/effects.py

```diff
@@ -0,0 +1,1180 @@
+			else:
+				location = util.get_cuser_location(context)
+
+			obj = add_area_particle_effect(context, effect, location)
+			if use_camera:
+				const = obj.constraints.new('COPY_LOCATION')
+				const.target = cam
+				const.use_z = False
+				obj.location = location
+
+		self.track_param = effect.name
+		return {'FINISHED'}
+
+
+class MCPREP_OT_instant_effect(bpy.types.Operator):
+	"""Spawn an effect that takes place at specific time and position"""
+	bl_idname = "mcprep.spawn_instant_effect"
+	bl_label = "Instant effect"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	def effects_enum(self, context):
+		"""Identifies eligible instant effects to include in the dropdown."""
+		mcprep_props = context.scene.mcprep_props
+		elist = []
+		for effect in mcprep_props.effects_list:
+			if effect.effect_type == COLLECTION:
+				display_type = "preanimated collection effect"
+			elif effect.effect_type == IMG_SEQ:
+				display_type = "image sequence effect"
+			else:
+				continue
+			elist.append((
+				str(effect.index),
+				effect.name,
+				"Add {} {} from {}".format(
+					effect.name,
+					display_type,
+					os.path.basename(effect.filepath))
+			))
+		return elist
+
+	effect_id = bpy.props.EnumProperty(items=effects_enum, name="Effect")
+	location = bpy.props.FloatVectorProperty(default=(0, 0, 0), name="Location")
+	frame = bpy.props.IntProperty(
+		default=0, name="Frame", description="Start frame for animation")
+	speed = bpy.props.FloatProperty(
+		default=1.0, min=0.1, name="Speed",
+		description="Make the effect run faster (skip frames) or slower (hold frames)")
+	show_image = bpy.props.BoolProperty(
+		default=True, name="Show image preview",
+		description="Show a middle animation frame as a viewport preview")
+	skipUsage = bpy.props.BoolProperty(default=False, options={'HIDDEN'})
+
+	@classmethod
+	def poll(cls, context):
+		return context.mode == 'OBJECT'
+
+	track_function = "effect_instant"
+	@tracking.report_error
+	def execute(self, context):
+		mcprep_props = context.scene.mcprep_props
+		effect = mcprep_props.effects_list[int(self.effect_id)]
+		if effect.effect_type == "collection":
+			add_collection_effect(
+				context, effect, self.location, self.frame)
+			return {'FINISHED'}
+		elif effect.effect_type == "img_seq":
+			inst = add_image_sequence_effect(
+				context, effect, self.location, self.frame, self.speed)
+
+			if not self.show_image:
+				inst.data = None
+		self.track_param = effect.name
+		return {'FINISHED'}
+
+
+class MCPREP_OT_spawn_particle_planes(bpy.types.Operator, ImportHelper):
+	"""Create a particle system from a selected image input"""
+	bl_idname = "mcprep.spawn_particle_planes"
+	bl_label = "Spawn Particle Planes"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	location = bpy.props.FloatVectorProperty(
+		default=(0, 0, 0), name="Location")
+	frame = bpy.props.IntProperty(default=0, name="Frame")
+
+	# Importer helper
+	filter_glob = bpy.props.StringProperty(
+		default="*.png;*.jpg;*.jpeg;*.tiff",
```

Comment:
I think it would be better to put the image formats in a global variable for easy modification

Reply:
Good idea, I've made the update now.